extensions/amp-ad-network-mgid-impl/0.1/amp-ad-network-mgid-impl.js

```diff
@@ -0,0 +1,293 @@
+
+      if (typeof networkInformation.type != 'undefined') {
+        params.push('nit=' + networkInformation.type);
+      }
+      if (typeof networkInformation.effectiveType != 'undefined') {
+        params.push('niet=' + networkInformation.effectiveType);
+      }
+      if (typeof networkInformation.saveData != 'undefined') {
+        params.push('nisd=' + (networkInformation.saveData ? 1 : 0));
+      }
+    } catch (e) {}
+
+    return params;
+  }
+
+  /**
+   * @return {string} Device pixel ratio info for ad request
+   * @private
+   */
+  getDevicePixelRatioParam_() {
+    let ratio = 1;
+
+    if (typeof window.devicePixelRatio !== 'undefined') {
+      ratio = window.devicePixelRatio;
+    } else if (
+      typeof window.screen.systemXDPI !== 'undefined' &&
+      typeof window.screen.logicalXDPI !== 'undefined' &&
+      window.screen.systemXDPI > window.screen.logicalXDPI
+    ) {
+      ratio = window.screen.systemXDPI / window.screen.logicalXDPI;
+    }
+
+    const isInt = ratio % 1 === 0;
+
+    if (!isInt) {
+      ratio = ratio.toFixed(3);
+    }
+
+    return 'dpr=' + ratio;
+  }
+
+  /**
+   * @return {string} Referrer info for ad request
+   * @private
+   */
+  getPrParam_() {
+    return this.getReferrer_(10).then((referrer) => {
+      const matchDomain = referrer.match(/:\/\/([^\/:]+)/i);
+      return (
+        'pr=' +
+        encodeURIComponent(matchDomain && matchDomain[1] ? matchDomain[1] : '')
+      );
+    });
+  }
+
+  /**
+   * @return {string} Current page url for ad request
+   * @private
+   */
+  getCxurlParam_() {
+    const url = Services.documentInfoForDoc(this.element).canonicalUrl;
+    return 'cxurl=' + encodeURIComponent(url);
+  }
+
+  /**
+   * @return {string} Pageview info for ad request
+   * @private
+   */
+  getPvidParam_() {
+    return Services.documentInfoForDoc(this.element).pageViewId64.then(
+      (pvid) => {
+        return 'pvid=' + pvid;
+      }
+    );
+  }
+
+  /**
+   * @return {string} Muidn info for ad request
+   * @private
+   */
+  getMuidParam_() {
+    return Services.urlReplacementsForDoc(this.element)
+      ./*OK*/ expandStringAsync('CLIENT_ID(muidn)', undefined, {
+        'CLIENT_ID': true,
+      })
+      .then((r) => {
+        return 'muid=' + r;
+      });
+  }
+
+  /**
+   * Returns the referrer or undefined if the referrer is not resolved
+   * before the given timeout
+   * @param {number=} opt_timeout
+   * @return {!(Promise<string>|Promise<undefined>)} A promise with a referrer or undefined
+   * if timed out
+   * @private
+   */
+  getReferrer_(opt_timeout) {
+    const timeoutInt = parseInt(opt_timeout, 10);
+    const referrerPromise = Services.viewerForDoc(
+      this.getAmpDoc()
+    ).getReferrerUrl();
+    if (isNaN(timeoutInt) || timeoutInt < 0) {
+      return referrerPromise;
+    }
+    return Services.timerFor(this.win)
+      .timeoutPromise(timeoutInt, referrerPromise)
+      .catch(() => undefined);
```

Comment:
Looks like another catch that is swallowing up errors. If this is intended, that's kind of odd, but we do a lot of odd things in AMP, so I guess it'll be fine. But if it's not intended then can we just remove the catch so that if there is an error, it just bubbles up like normal?

Reply:
It's intended.